src/legacy/ui/public/agg_types/controls/ranges.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Fragment, useState, useEffect } from 'react';
+import {
+  htmlIdGenerator,
+  EuiButtonIcon,
+  EuiFieldNumber,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiIcon,
+  EuiSpacer,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import { isEqual, omit } from 'lodash';
+import { AggParamEditorProps } from 'ui/vis/editors/default';
+
+const generateId = htmlIdGenerator();
+const isEmpty = (value: any) => value === undefined || value === null;
+
+interface RangeValues {
+  from?: number;
+  to?: number;
+}
+
+interface RangeValuesModel extends RangeValues {
+  id: string;
+}
+
+function RangesParamEditor({ agg, value = [], setValue }: AggParamEditorProps<RangeValues[]>) {
+  const [ranges, setRanges] = useState(() => value.map(range => ({ ...range, id: generateId() })));
```

Comment:
Not sure you need to use a callback state here, couldn't you write: `const [ranges, setRanges] = useState(value.map(...));`

Reply:
I think the lazy init is used to prevent an id generation on every render.

Reply:
@flash1293 you are completely right